### PR TITLE
Add Android 11 permissions for getPhoneNumber

### PR DESCRIPTION
**Pull request by colbygarland**
## Description

This is just adding the new permission to the README for `getPhoneNumber`. Note that I only added the permission that worked for me, there is another permission (linked in #1105) that worked for somebody else. But this was an old ticket, and this solution didn't work for me, so I did not add this permission. If we want, I can definitely add that one too!

Relates to #1364, #1105

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |
| Windows |    ✅     |

## Checklist

* [x] I have tested this on a device/simulator for each compatible OS
* [x] I added the documentation in `README.md`
* [ ] I updated the typings files (`privateTypes.ts`, `types.ts`)
* [ ] I added a sample use of the API (`example/App.js`)

### CODE REVIEW

**Review by schie — APPROVED** [review]
LTGM

**Review by mikehardy — COMMENTED** [review]
I really appreciate this PR, as the old permission will not work, but I admit I am very confused why READ_SMS worked for you, but the read phone numbers one did not? That just seems contrary to the documentation. I don't have time to personally test though, and I believe your actual testing experience. I just worry because the documentation for READ_SMS lists it as "hard restricted"  https://developer.android.com/reference/android/Manifest.permission#READ_SMS

What would make me more comfortable is if the permissions discussion here was a little less prescriptive now and linked back to the actual documentation listing how it is supposed to work: https://developer.android.com/about/versions/11/privacy/permissions#phone-numbers - along with a note that READ_SMS may work as well, if the documented permissions aren't successful.

Basically, refer people to upstream docs, but also include your experience. What do you think?

**Comment by colbygarland** [comment]
@mikehardy okay yeah I totally agree. I was hesitant to even create this PR, since my testing was so different than others.. _but_ if we just link to the upstream docs, and I add a disclaimer about what worked for me, that might be the best solution. With these "dangerous" permissions, we should probably put more information then less, since they are not only hard to get users to accept, but also get accepted by the App stores.

**Review by mikehardy — APPROVED** [review]
I think this is the best we (that is: you, thanks for the PR :-) ) can do. It's not great to just expose the mess to users, but there's not a better way I think, especially with permissions that are tricky like this (and require disclosure on the store listing and everything). LGTM
